src/script/conversation/ConversationRepository.ts

```diff
@@ -2394,14 +2394,6 @@ export class ConversationRepository {
 
     const creatorIsParticipant = createdByParticipant || createdBySelfUser;
 
-    const data = await this.conversationService.getConversationById(conversationEntity);
-    const allMembers = [...data.members.others, data.members.self];
-    const conversationRoles = allMembers.reduce<Record<string, string>>((roles, member) => {
-      roles[member.id] = member.conversation_role;
-      return roles;
-    }, {});
-    conversationEntity.roles(conversationRoles);
-
```

Comment:
This is where we fetched the conversation for a second time and then mapped the roles (this is now done in the `ConversationMapper`)